scripts/zones/AlTaieu/mobs/Absolute_Virtue.lua

```diff
@@ -4,20 +4,274 @@
+}
+
+entity.isLocked = function(sp)
+    if #xi.av.sps == 0 then
+        return true
+    end
+
+    for _, jsa in ipairs(xi.av.sps) do
+        if sp == jsa then
+            return false
+        end
+    end
+
+    return true
+end
+
+entity.lock = function(sp)
+    for i, jsa in ipairs(xi.av.sps) do
+        if jsa == sp then
+            debug(string.format('locked: %d', sp))
+            table.remove(xi.av.sps, i)
+            break
+        end
+    end
+
+    for i, jsa in ipairs(xi.av.braceletsps) do
+        if jsa == sp then
+            table.remove(xi.av.braceletsps, i)
+            break
+        end
+    end
+end
+
+-- TODO: handle pets and pet sp abilities
+entity.handleSP = function(mob)
+    local now = os.time()
+    if now > xi.av.nextsp then
+        if xi.av.bracelets and #xi.av.braceletsps ~= 0 then
+            local trigger = xi.av.braceletsps[math.random(1, #xi.av.braceletsps)]
+            local combo = combos[trigger]
+            if trigger == xi.jsa.CHAINSPELL then
+                combo = combos[math.random(1, 2)]
+            end
+
+            for _, jsa in ipairs(combo) do
+                if not entity.isLocked(jsa) then
+                    mob:setLocalVar(string.format('sp_%u', jsa), os.time())
+                    debug(string.format('%s using %d', mob:getName(), jsa))
+                    mob:useMobAbility(jsa)
+                end
+            end
+        elseif #xi.av.sps ~= 0 then
+            local sp = xi.av.sps[math.random(1, #xi.av.sps)]
+            mob:setLocalVar(string.format('sp_%u', sp), os.time())
+            debug(string.format('%s using %d', mob:getName(), sp))
+            mob:useMobAbility(sp)
+        end
+
+        -- TODO: minimum should be max of 2hr combos length
+        xi.av.nextsp = now + math.random(45, 90)
+    end
+end
+
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.ABILITY_RESPONSE, 1)
+
+    --[[
+    mob:setMobMod(xi.mobMod.MAGIC_COOL, 20)
+    mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180)
+    mob:setMod(xi.mod.SOULEATERRES, 4)
+    mob:setMod(xi.mod.UFASTCAST, 100)
+    mob:setMod(xi.mod.MAIN_DMG_RATING, 150)
+    mob:setMod(xi.mod.ATT, 300)
+    mob:setMod(xi.mod.DEF, 300)
+    mob:setMod(xi.mod.MATT, 132)
+    mob:setMod(xi.mod.MDEF, 200)
+    mob:setMod(xi.mod.REFRESH, 500)
+    mob:setMod(xi.mod.REGAIN, 200)
+
+    mob:addImmunity(xi.immunity.SLEEP)
+    mob:addImmunity(xi.immunity.GRAVITY)
+    mob:addImmunity(xi.immunity.BIND)
+    mob:addImmunity(xi.immunity.STUN)
+    mob:addImmunity(xi.immunity.SILENCE)
+    mob:addImmunity(xi.immunity.PARALYZE)
+    mob:addImmunity(xi.immunity.BLIND)
+    mob:addImmunity(xi.immunity.SLOW)
+    mob:addImmunity(xi.immunity.POISON)
+    mob:addImmunity(xi.immunity.ELEGY)
+    mob:addImmunity(xi.immunity.REQUIEM)
+
+    mob:speed(60)
+    ]]--
+end
+
 entity.onMobSpawn = function(mob)
-    -- setMod
-    mob:setMod(xi.mod.REGEN, 500)
+    -- reset av
+    mob:setAnimationSub(1)
+
+    xi.av.regen = 250
```

Comment:
I don't remember the details but thought the regen amounts were confirmed ages ago

Reply:
I fully have no idea 🤷